inference-engine/thirdparty/clDNN/tests/test_cases/reorder_gpu_test.cpp

```diff
@@ -2396,6 +2399,214 @@ INSTANTIATE_TEST_SUITE_P(DISABLED_REORDER,
+    );
+
+    execute(p);
+
+    EXPECT_EQ(check_optimized_out(p, "reorder_bias1"), true);
+}
+
+// Testing bugfix not to remove reorder in front of conv has deep depth input
+TEST_P(testing_removal_reorder, only_remove_reorder_shallow_depth_input) {
+    auto p = GetParam();
+    layout reorder_layout(data_types::u8, format::b_fs_yx_fsv32, p.in_shape, padding({0, }, 0));
+
+    create_topologies(input_layout("input", get_input_layout(p)),
+        data("weights", get_mem(get_weights_layout(p))),
+        data("bias", get_mem(get_bias_layout(p))),
+        data("weights_sec", get_mem(get_weights_layout(p))),
+        reorder("reorder_fp32", "input", format::bfyx, data_types::f32),
+        convolution("conv_prim", "reorder_fp32", { "weights" }, { "bias" }, 1, p.stride, p.pad, tensor{1}, p.in_shape, data_types::u8, false),
+        reorder("reorder_conv", "conv_prim", reorder_layout),
+        convolution("conv_output", "reorder_conv", { "weights_sec" }, 1, p.stride, p.pad),
+        reorder("reorder_bfyx", "conv_output", format::b_fs_yx_fsv32, data_types::f32),
+        resample("resample", "reorder_bfyx", p.out_shape, 1),
+        reorder("reorder_output", "resample", p.default_format, data_types::f32)
+    );
+
+    execute(p);
+
+    EXPECT_EQ(check_optimized_out(p, "reorder_conv"), false);
+}
+
+#ifdef ENABLE_ONEDNN_FOR_GPU
+// Check to remove reorder between onednn and cldnn conv if the reorder has no padded output
+TEST_P(testing_removal_reorder, removal_no_padded_reorder) {
+    auto p = GetParam();
+    layout reorder_layout(data_types::f16, format::b_fs_yx_fsv16, p.in_shape, padding({0, }, 0));
+
+    create_topologies(input_layout("input", get_input_layout(p)),
+        data("weights", get_mem(get_weights_layout(p))),
+        data("bias", get_mem(get_bias_layout(p))),
+        reorder("reorder_fp32", "input", format::bfyx, data_types::f16),
+        convolution("conv_prim", "reorder_fp32", { "weights" }, 1, p.stride, p.pad),
+        reorder("reorder_conv", "conv_prim", reorder_layout),
+        convolution("conv_output", "reorder_conv", { "weights" }, 1, p.stride, p.pad),
+        reorder("reorder_output", "conv_output", p.default_format, data_types::f32)
+    );
+
+    auto build_opts = build_options();
+    build_opts.set_option(build_option::optimize_data(true));
+    implementation_desc impl = { format::b_fs_yx_fsv16, std::string(""), impl_types::ocl };
+    build_opts.set_option(build_option::force_implementations({ {"conv_output", impl} }));
+
+    setup_with_build_ops(build_opts);
+
+    execute(p);
+
+    if (check_supports_immad())
+        EXPECT_EQ(check_optimized_out(p, "reorder_conv"), true);
+    else
+        EXPECT_EQ(check_optimized_out(p, "reorder_conv"), false);
+}
+
+// Check not to remove reorder between onednn and cldnn conv if the reorder has padded output
+TEST_P(testing_removal_reorder, removal_padded_reorder) {
+    auto p = GetParam();
```

Comment:
I think you can make it explicit that this is only for onednn test, by adding code below:
```
if (!check_supports_immad())
    return;
```

Reply:
Applied

Reply:
I mean, `removal_padded_reorder` is also for onednn only. isn't it?